src/telemetry.ts

```diff
@@ -3174,28 +3190,100 @@ export class IEventNamePropertyMapping {
+    };
+    /**
+     * An event describing whether the environment has ipykernel 6 installed.
+     */
     [DebuggingTelemetry.ipykernel6Status]: TelemetryEventInfo<{
         status: 'installed' | 'notInstalled';
-    }> = { owner: 'roblourens' } as any;
-    [DebuggingTelemetry.clickedRunByLine]: TelemetryEventInfo<never | undefined> = { owner: 'roblourens' } as any;
-    [DebuggingTelemetry.successfullyStartedRunByLine]: TelemetryEventInfo<never | undefined> = {
-        owner: 'roblourens'
-    } as any;
+    }> = {
+        owner: 'roblourens',
+        feature: ['Debugger'],
+        properties: {
+            /**
+             * Whether ipykernel 6 is installed.
+             */
+            status: {
+                classification: 'SystemMetaData',
+                purpose: 'FeatureInsight'
+            }
+        },
+        source: 'N/A'
+    };
+    /**
+     * Sent when the user attempts to start run by line.
+     */
+    [DebuggingTelemetry.clickedRunByLine]: TelemetryEventInfo<never | undefined> = {
+        owner: 'roblourens',
+        feature: ['Debugger'],
+        source: 'User Action'
+    };
+    /**
+     * Sent when the run by line session starts successfully.
+     */
+    [DebuggingTelemetry.successfullyStartedRunByLine]: TelemetryEventInfo<undefined> = {
+        owner: 'roblourens',
+        feature: ['Debugger'],
+        source: 'N/A'
+    };
     /**
      * Telemetry sent when we have managed to successfully start the Interactive Window debugger using the Jupyter protocol.
      */
     [DebuggingTelemetry.successfullyStartedIWJupyterDebugger]: TelemetryEventInfo<never | undefined> = {
-        owner: 'roblourens'
-    } as any;
-    [DebuggingTelemetry.clickedRunAndDebugCell]: TelemetryEventInfo<never | undefined> = { owner: 'roblourens' } as any;
+        owner: 'roblourens',
+        feature: ['Debugger'],
+        source: 'N/A'
+    };
+    /**
+     * Sent when the user attempts to start debugging a notebook cell.
+     */
+    [DebuggingTelemetry.clickedRunAndDebugCell]: TelemetryEventInfo<never | undefined> = {
+        owner: 'roblourens',
+        feature: ['Debugger'],
+        source: 'User Action'
+    };
+    /**
+     * Sent when the user successfully starts debugging a notebook cell.
+     */
     [DebuggingTelemetry.successfullyStartedRunAndDebugCell]: TelemetryEventInfo<never | undefined> = {
-        owner: 'roblourens'
-    } as any;
+        owner: 'roblourens',
+        feature: ['Debugger'],
+        source: 'N/A'
+    };
+    /**
+     * Sent when a notebook debugging session ends.
+     */
     [DebuggingTelemetry.endedSession]: TelemetryEventInfo<{
         reason: 'normally' | 'onKernelDisposed' | 'onAnInterrupt' | 'onARestart' | 'withKeybinding';
-    }> = { owner: 'roblourens' } as any;
+    }> = {
+        owner: 'roblourens',
+        feature: ['Debugger'],
+        source: 'N/A',
+        properties: {
+            /**
+             * The reason the session ended.
```

Comment:
You need this comment on the property in line 3272.
I.e. add jsDoc comments to the property itself.